app/controllers/IndexController.scala

```diff
@@ -47,17 +48,25 @@ class IndexController @Inject() (
       val fatcaId = request.fatcaId
       subscriptionService.getSubscription(fatcaId).flatMap {
         sub =>
-          val changeContactDetailsUrl = if (sub.isBusiness) conf.changeOrganisationDetailsUrl else conf.changeIndividualDetailsUrl
-          val addNewFIUrl             = controllers.addFinancialInstitution.routes.AddFIController.onPageLoad.url
+          val changeContactDetailsUrl       = if (sub.isBusiness) conf.changeOrganisationDetailsUrl else conf.changeIndividualDetailsUrl
+          val addNewFIUrl                   = "/manage-your-crs-and-fatca-financial-institutions/add"
+          val hasFisFuture: Future[Boolean] = financialInstitutionsService.getListOfFinancialInstitutions(fatcaId).map(_.nonEmpty)
 
-          sessionRepository.get(request.userId) flatMap {
-            case Some(_) => Future.successful(Ok(view(sub.isBusiness, sub.businessName, fatcaId, addNewFIUrl, changeContactDetailsUrl)))
-            case None =>
-              sessionRepository.set(UserAnswers(request.userId)) map {
-                case true => Ok(view(sub.isBusiness, sub.businessName, fatcaId, addNewFIUrl, changeContactDetailsUrl))
-                case false =>
-                  logger.error(s"Failed to initialize user answers for userId: [${request.userId}]")
-                  Redirect(controllers.routes.JourneyRecoveryController.onPageLoad())
+          hasFisFuture.flatMap {
+            hasFis =>
+              val indexPageDetails =
+                IndexViewModel(sub.isBusiness, fatcaId, addNewFIUrl, changeContactDetailsUrl, sub.businessName.getOrElse(""), hasFis)
```

Comment:
If the business name is missing would the empty string still be meaningful from the design perspective?

Reply:
it would be hidden behind a Boolean so wouldnt ever appear unless isBusiness=true